### ISSUE TITLE
Main Branch Build Failure in Forks Due to Fork Repo Name

### ISSUE THREAD

**Issue by iamjpotts** [ansible-collections/community.docker#502]
##### SUMMARY

After syncing my fork's main with the source repository to pull in yesterday's commits, I now get a build failure that appears to be due to the name of my fork.

[Link to failing build](https://github.com/iamjpotts/ansible-collections-community.docker/actions/runs/3558790142/jobs/5977708279)

`ERROR! Invalid collection name (must be of the form namespace.collection): ansible-collections-community.docker`

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

Component: git fork of repo

##### ANSIBLE VERSION

```paste below
ansible [core 2.13.6]
  config file = None
  configured module search path = ['~/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = ~/dev/ansible/collections/venv/lib/python3.8/site-packages/ansible
  ansible collection location = ~/.ansible/collections:/usr/share/ansible/collections
  executable location = ~/dev/ansible/collections/venv/bin/ansible
  python version = 3.8.10 (default, Jun 22 2022, 20:18:18) [GCC 9.4.0]
  jinja version = 3.1.2
  libyaml = True
```

##### COLLECTION VERSION

Collection version: current `HEAD` of this repository

##### CONFIGURATION

GitHub CI workflow

##### OS / ENVIRONMENT

GitHub CI workflow

##### STEPS TO REPRODUCE

Run the GitHub CI workflow

##### EXPECTED RESULTS

Build succeeds regardless of name of repository fork

##### ACTUAL RESULTS

Build fails

**Comment by felixfontein** [comment]
A general note: I always disable GHA on forks, since I've never seen a case so far where running the actions in your fork is useful.

Right now we're using the repository name as the default for the collection's name (https://github.com/ansible-community/github-docs-build/blob/main/.github/workflows/_shared-docs-build-push.yml#L10). This usually is no problem as folks usually don't rename their forks. Changing this requires us to hard-code the repository name in the GH workflow, which makes it harder to synchronize the workflow between multiple collection repos. But then the name is already in there (in the `if` for the publish step), so it won't be that bad.

CC @briantist

**Comment by felixfontein** [comment]
Fixed in d043fc6cbc81d6611d843ccd1d2c53f1b7cb5417.

**Comment by briantist** [comment]
yeah, we're in a tough spot with this one. It has to be hardcoded in the `if` due to GHA limitations, I would have preferred not to at all. Renamed forks are far less common, so I'd rather keep things the way they are and let collection maintainers decide if they want to accommodate those forks the way it was done here. And as mentioned, forkers have the option of disabling the workflows in their fork without modifying any of the files.
